analyzers/tests/SonarAnalyzer.UnitTest/Rules/SymbolicExecution/ConditionEvaluatesToConstantTest.cs

```diff
@@ -72,6 +74,45 @@ public class ConditionEvaluatesToConstantTest
             Verifier.VerifyAnalyzerFromCSharp9Console(@"TestCases\ConditionEvaluatesToConstant.CSharp9.TopLevelStatements.cs", GetAnalyzer());
 #endif
 
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void ConditionEvaluatesToConstant_UncaughtException()
+        {
+            Action action = () => Verifier.VerifyCSharpAnalyzer(@"
+using System;
+public class Reproducer
+{
+    private DateTime? foo;
+
+    public virtual DateTime? Foo
+    {
+        get { return foo; }
+
+        set
+        {
+            if (value.HasValue)
+            {
+                value = DateTime.SpecifyKind(value.Value, DateTimeKind.Unspecified);
+            }
+
+            if (foo != value || (foo.HasValue && value.HasValue && foo.Value.Kind != value.Value.Kind))
+            {
+                foo = value;
+                Bar(""x"");
+            }
+        }
+    }
+    public void Bar(string x) { }
+}", GetAnalyzer(), CompilationErrorBehavior.Ignore);
+
+            // https://github.com/SonarSource/sonar-dotnet/issues/4573
+            action.Should().Throw<AssertFailedException>().Where(e =>
+                e.Message.Contains("Expected diagnostics {error AD0001: Analyzer 'SonarAnalyzer.Rules.SymbolicExecution.SymbolicExecutionRunner' threw an exception of type " +
+                "'SonarAnalyzer.SymbolicExecution.SymbolicExecutionException' with message 'Error processing method: set_Foo ## Method file: snippet1.cs " +
```

Comment:
FYI Dima, because Roslyn keeps only the first line of an exception message, we concatenate all the lines in a single one where we replace newline by `##`. Like this, we can copy-paste the line locally and replace `##` with newline and have a beautiful stacktrace when examining some issue.